src/test/ui/consts/static-ice.rs

```diff
@@ -0,0 +1,27 @@
+// check-pass
+
+#[derive(Copy, Clone)]
+pub struct Glfw;
+
+static mut GLFW: Option<Glfw> = None;
+pub fn new() -> Glfw {
+    unsafe {
+        if let Some(glfw) = GLFW {
+            return glfw;
+        } else {
+            todo!()
+        }
+    };
+}
+
+extern "C" {
+    static _dispatch_queue_attr_concurrent: [u8; 0];
+}
+
+static DISPATCH_QUEUE_CONCURRENT: &'static [u8; 0] =
+    unsafe { &_dispatch_queue_attr_concurrent };
+
+fn main() {
+    *DISPATCH_QUEUE_CONCURRENT;
+    new();
+}
```

Comment:
Also maybe could you also add a testcase from https://github.com/rust-lang/rust/issues/71709? The ICE messages are quite different, so better safe than sorry in terms of test coverage...

Reply:
This test tests both issues at the same time.